build.gradle.kts

```diff
@@ -50,3 +77,83 @@ subprojects {
         }
     }
 }
+
+/**
+ * To run detekt simply:
+ * 1. ./gradlew module:detekt for each module
+ * 2. ./gradlew detekt for whole project
+ */
+fun Project.coreDetektSetup() {
+    // Apply Plugin to sub-project
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+
+    // Configure Detekt
+    detekt {
+        config = files("$rootDir/config/detekt/detekt.yml")
+        buildUponDefaultConfig = true
+        ignoredBuildTypes = listOf("release")
```

Comment:
wouldn't we want to run this on all builds? why ignore `release`?

Reply:
I'm assuming detekt should way before you want a release buid. It would add more overhead on release builds. But we can enable it if that's not a big deal

Reply:
Yeah, if all the branching and release strategy is done right, technically it shouldn't be needed. But if we end up doing nightly release builds for internal testing or a beta or something, might be good to have it. Regardless, no changes needed! We can cross that bridge when we get there, just wanted to call this out.

Nice Work!